File: content-solutions/greedy/fewest-jumps.js
function solution(jumps) {
  let count = 0;
  let end = 0;
  let far = 0;
  for (let i = 0; i < jumps.length - 1; i++) {
    far = Math.max(far, i + jumps[i]);
    if (i === end) {
      count++;
      end = far;
    }
  }
  return count;
}

File: content-solutions/greedy/fewest-jumps.java
class Solution {
    public int solution(int[] jumps) {
        int count = 0, end = 0, far = 0;
        for (int i = 0; i < jumps.length - 1; i++) {
            far = Math.max(far, i + jumps[i]);
            if (i == end) {
                count++;
                end = far;
            }
        }
        return count;
    }
}
